From 73b31f5e57e11f9e60744e37800f87842f183452 Mon Sep 17 00:00:00 2001
From: Kweku Adams <kwekua@google.com>
Date: Thu, 3 Jun 2021 16:07:56 -0700
Subject: [PATCH] Don't export HeapDumpProvider.

Stop exporting HeapDumpProvider so apps can only access generated dumps
when the user explicitly shares them.

Bug: 184046948
Test: capture system heap dump in developer options and confirm test app
get SecurityException if it tries to access the dump directly, but gets
access when the dump is shared through the notification flow

Change-Id: Ibdca7cde4f563baa39163869289da5b79fc3a6db
(cherry picked from commit a60c62bcb74e0146820f75f1da49581d1709b63c)
(cherry picked from commit 7d6e3a0b24a53529532e6a0c14688f0681e44b62)
---
 packages/Shell/AndroidManifest.xml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/packages/Shell/AndroidManifest.xml b/packages/Shell/AndroidManifest.xml
index 570c2ab3efd1..3e820c39665c 100644
--- a/packages/Shell/AndroidManifest.xml
+++ b/packages/Shell/AndroidManifest.xml
@@ -346,7 +346,7 @@
         <provider android:name=".HeapDumpProvider"
                   android:authorities="com.android.shell.heapdump"
                   android:grantUriPermissions="true"
-                  android:exported="true" />
+                  android:exported="false" />
 
         <activity
             android:name=".BugreportWarningActivity"
-- 
2.17.1

